test/links.c

```diff
@@ -9890,16 +9891,21 @@ external_set_elink_cb(hid_t fapl, bool new_format)
     /* Build user data for callback */
     op_data.parent_file = filename1;
     op_data.target_file = filename2;
+
+    /* Check if using a parallel file driver */
+    if (h5_using_parallel_driver(fapl, &driver_is_parallel) < 0)
+        TEST_ERROR;
+
+    base_driver = H5Pget_driver(fapl);
+
     /* Core file driver has issues when used as the member file driver for a family file */
     /* Family file driver cannot be used with family or multi drivers for member files */
     /* Also disable parallel member drivers, because H5F_HAS_FEATURE(H5FD_FEAT_HAS_MPI)
        would report false, causing problems */
-    base_driver = H5Pget_driver(fapl);
-    op_data.base_fapl =
-        (base_driver == H5FD_FAMILY || base_driver == H5FD_MULTI || base_driver == H5FD_MPIO ||
-         base_driver == H5FD_CORE || base_driver == H5FD_DIRECT || base_driver == H5FD_SUBFILING)
-            ? H5P_DEFAULT
-            : fapl;
+    op_data.base_fapl = fapl;
+    if (base_driver == H5FD_CORE || base_driver == H5FD_FAMILY || base_driver == H5FD_MULTI ||
+            base_driver == H5FD_DIRECT || driver_is_parallel)
```

Comment:
With the rework of the library initialization, these VFD macros are mapped to an initialization call now instead of just an ID value. Because of that, this code initializes the Subfiling VFD when it's enabled and that seems to cause issues on Theta. This should resolve the failure at https://my.cdash.org/test/98835800 without causing an initialization of the VFD, while still basically checking for the same conditions as before (except that any future parallel VFDs may need an exception in this test, but it should probably be reworked anyway).

Reply:
We really need to rethink library initialization, especially when it comes to VFDs.